schema/resolvers.test.js

```diff
@@ -225,4 +404,343 @@ describe("resolvers", () => {
+          res => res.data.createRoutingCondition
+        );
+
+      beforeEach(async () => {
+        const routingRuleSet = await createNewRoutingRuleSet(firstPage.id).then(
+          res => res.data.createRoutingRuleSet
+        );
+        input = {
+          comparator: "Equal",
+          questionPageId: firstPage.id,
+          routingRuleId: get(routingRuleSet, "routingRules[0].id")
+        };
+      });
+
+      it("should create a routing condition for question page", async () => {
+        const newRoutingCondition = await newCondition(input);
+        expect(newRoutingCondition).toMatchObject({
+          questionPage: {
+            id: firstPage.id
+          }
+        });
+      });
+
+      it("should delete a routing condition for question page", async () => {
+        const newRoutingCondition = await newCondition(input);
+        const page = await getRoutingDataForPage(firstPage);
+        expect(
+          get(page, "routingRuleSet.routingRules[0].conditions")
+        ).toHaveLength(2);
+
+        await deleteRoutingConditionMutation({ id: newRoutingCondition.id });
+
+        const updated = await getRoutingDataForPage(firstPage);
+        expect(
+          get(updated, "routingRuleSet.routingRules[0].conditions")
+        ).toHaveLength(1);
+      });
+
+      it("should set condition to first answer in page if one exists", async () => {
+        const answer = await createNewAnswer(firstPage, "Checkbox");
+        const routingCondition = await newCondition(input);
+
+        expect(routingCondition).toMatchObject({
+          answer: {
+            id: answer.id
+          }
+        });
+      });
+
+      it("should always use first answer on page when creating a condition", async () => {
+        const answer1 = await createNewAnswer(firstPage, "Number");
+        await createNewAnswer(firstPage, "Checkbox");
+        const routingCondition = await newCondition(input);
+
+        expect(routingCondition).toMatchObject({
+          answer: {
+            id: answer1.id
+          }
+        });
+      });
+
+      it("should be possible to change the page for the condition", async () => {
+        const section = first(sections);
+        const page = await getRoutingDataForPage(firstPage);
+
+        const routingCondition = get(
+          page,
+          "routingRuleSet.routingRules[0].conditions[0]"
+        );
+        expect(routingCondition).toMatchObject({
+          questionPage: {
+            id: firstPage.id
+          }
+        });
+
+        const newPage = await addPage(section.id);
+
+        await changeRoutingConditionMutation({
+          id: routingCondition.id,
+          questionPageId: newPage.id
+        });
+
+        const updated = await getRoutingDataForPage(firstPage);
+        expect(
+          get(updated, "routingRuleSet.routingRules[0].conditions[0]")
+        ).toMatchObject({
+          questionPage: {
+            id: newPage.id
+          }
+        });
```

Comment:
Possibly want a test here somewhere to make sure two condition values with the same args can't be inserted?